WooCommerce/Classes/ViewRelated/Themes/ThemesCarouselViewModel.swift

```diff
@@ -0,0 +1,111 @@
+import Foundation
+import Yosemite
+
+/// View model for `ThemesCarouselView`
+///
+final class ThemesCarouselViewModel: ObservableObject {
+
+    @Published private(set) var state: State = .loading
+    @Published private var themes: [WordPressTheme] = []
+    @Published private var currentThemeID: String?
+
+    let mode: Mode
+    private let stores: StoresManager
+
+    init(mode: Mode, stores: StoresManager = ServiceLocator.stores) {
+        self.mode = mode
+        self.stores = stores
+        observeThemeListAndCurrentTheme()
```

Comment:
❓ Would it make sense to observe only if `mode` is `themeSettings`? This is to avoid setting `state` from two places for `storeCreationProfiler` mode. Please correct me if I am wrong.

Reply:
Good idea! Updated in a043a7a0a39868c00ca5b2092d108bd1655027e5.